caput/tests/test_mpiarray.py

```diff
@@ -289,14 +293,18 @@ def test_global_getslice(self):
         res = local_array[:, 3:5]
 
         assert isinstance(arr, mpiarray.MPIArray)
+        assert arr.axis == 0
```

Comment:
This file seems to be an awkward mix of pytest (e.g. bare asserts) and unittest (e.g. self.assertRaisesWhenTheMoonIsFull) style test cases. I think the former supports the latter but not the other way around, so I'd probably recommend converting the entire file to pytest style.

There don't appear to be any setup/teardown methods so it should be very quick to port as you don't need to write any pytest fixtures.

Reply:
I was using the `unittest.assertRaises` to be able to test that an exception occurred when it was supposed to. I want to check if pytest has something that does something similar.

Reply:
Examples:

```
         # check that subtracting arrays with two different distributed axis fails                                                                                                                              
         self.assertRaises(                                                                                                                                                                                     
             mpiarray.AxisException,                                                                                                                                                                            
             np.subtract,                                                                                                                                                                                       
             mpiarray.MPIArray((size, 4), axis=0),                                                                                                                                                              
             mpiarray.MPIArray((size, 4), axis=1),                                                                                                                                                              
         )                                                                                                                                                                                                      
                                                                                                                                                                                                                
         # check that outer ufunc on arrays that cannot be broadcast fails                                                                                                                                      
         self.assertRaises(                                                                                                                                                                                     
             ValueError,                                                                                                                                                                                        
             np.multiply,                                                                                                                                                                                       
             mpiarray.MPIArray((size, 3), axis=0),                                                                                                                                                              
             mpiarray.MPIArray((size, 4), axis=0),                                                                                                                                                              
         )                                                                                                                                                                                                      
```

Reply:
It seems it does support it: https://docs.pytest.org/en/stable/assert.html

Reply:
Indeed it does. I think you mostly need to change that and then change all the class methods into functions.